src/MarketingPageAcceptanceTestsSpecflow/Utils/UITest.cs

```diff
@@ -25,10 +25,11 @@ public sealed class UITest
         
         public UITest()
         {
-            solution = CreateSolution.CreateNewSolution();
-            solutionDetail = CreateSolutionDetails.CreateNewSolutionDetail(solution.Id, Guid.NewGuid(), 0, false);
+            connectionString = EnvironmentVariables.GetConnectionString();
 
-            connectionString = EnvironmentVariables.GetConnectionString();            
+            solution = CreateSolution.CreateNewSolution(checkForUnique: true, connectionString: connectionString);
```

Comment:
since checkForUnique is true for every test run, why not just have it to true as default?

Reply:
It allows for the data generator to create randomised solutions for other uses (i.e. if the API guys end up using it for unit tests) that don't need the database entry adding